### ISSUE TITLE
DAY/YEAR triggers L014 in DATE_DIFF, LAST_DAY in BigQuery

### ISSUE THREAD

**Issue by theianrobertson** [sqlfluff/sqlfluff#528]
SQL that I don't expect to throw any capitalization rules:

```sql
SELECT
    LAST_DAY(date_field, YEAR) AS year_end_date,
    DATE_DIFF(date_field, date_field_2, DAY) AS date_diff_field,
    EXTRACT(YEAR FROM date_field) AS day_year,
    DATE_ADD(date_field, INTERVAL 1 DAY) AS next_day
FROM
    dates
```

Running lint with BigQuery dialect gives me:

```
L:   2 | P:  26 | L014 | Inconsistent capitalisation of unquoted identifiers.
L:   3 | P:  41 | L014 | Inconsistent capitalisation of unquoted identifiers.
```

`YEAR` and `DAY` are OK in the `EXTRACT` and `DATE_ADD` functions (line 4/5), but not OK in the `LAST_DAY` and `DATE_DIFF` functions (line 2/3).  I'd expect them to be treated the same way (my preference is capitalize all)

**Comment by NiallRees** [comment]
Relates to #443

**Comment by barrywhart** [comment]
This issue is closely related to #357, in that both involve SQLFluff being "confused" by BigQuery's occasional use of certain keywords as function parameters. See that issue for some discussion of potential designs for addressing this.

**Comment by barrywhart** [comment]
I tested this again, and it no longer reports an L014 violation.  I think it was resolved by #708. Closing.
